packages/editor/src/lib/editor/derivations/bindingsIndex.ts

```diff
@@ -0,0 +1,89 @@
+import { Computed, RESET_VALUE, computed, isUninitialized } from '@tldraw/state'
+import { TLBinding, TLShapeId } from '@tldraw/tlschema'
+import { objectMapValues } from '@tldraw/utils'
+import { Editor } from '../Editor'
+
+type TLBindingsIndex = Record<TLShapeId, undefined | TLBinding[]>
```

Comment:
happy to call it out of scope of this diff as we use objects as maps _a lot_ in tldraw, but i believe that this pattern (using an object as a map instead of a `Map` or other ds) is a big perf foot-gun in JS as it forces V8/other engines to constantly create & throw out the hidden classes that back the objects. thoughts on using a normal map here?

Reply:
i switched to a map. would be fun to do a perf comparison but I'm lazy.

yeah i'd be interested in an immutable map too. Immutable.js would be way too heavy a dependency but that red-black tree implementation looks cool, wasn't aware of it.

Reply:
yeahhhh, i think there's a version of immutable with just the map? but not sure